Builds/CMake/CMakeFuncs.cmake

```diff
@@ -502,6 +498,15 @@ macro(setup_build_boilerplate)
       APPEND
       PROPERTY COMPILE_DEFINITIONS
       $<$<OR:$<BOOL:${profile}>,$<CONFIG:Release>,$<CONFIG:ReleaseClassic>>:NDEBUG>)
+  else()
+    # CMAKE_CXX_FLAGS_RELEASE is created by CMake for most / all generators
+    #  with defaults including /DNDEBUG or -DNDEBUG, and that value is stored
+    #  in the cache. Override that locally so that the cache value will be
+    #  avaiable if "assert" is ever changed.
+    STRING(REGEX REPLACE "[-/][A-Z]*NDEBUG" "" CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
+    STRING(REGEX REPLACE "[-/][A-Z]*NDEBUG" "" CMAKE_CXX_FLAGS_RELEASECLASSIC "${CMAKE_CXX_FLAGS_RELEASECLASSIC}")
+    STRING(REGEX REPLACE "[-/][A-Z]*NDEBUG" "" CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
+    STRING(REGEX REPLACE "[-/][A-Z]*NDEBUG" "" CMAKE_C_FLAGS_RELEASECLASSIC "${CMAKE_C_FLAGS_RELEASECLASSIC}")
```

Comment:
You're right, they aren't lists (I'm surprised, but good catch).

I am in favor of changing the regex. The risks we are weighing are: 1) using a compiler that uses something other than `[-/]D` and 2) mistakenly changing a define that happens to end in `NDEBUG`. I think (2) is the bigger risk, but I'm fine either way.